contrib/babelfishpg_common/src/spatialtypes.c

```diff
@@ -67,134 +113,1465 @@ load_functions()
 
-        geom_datum = gserialized_set_srid_p(fcinfo_local);
-    }
-    geometry_name = text_to_cstring(PG_DETOAST_DATUM(geom_type));
+    geometry_name = text_to_cstring(DatumGetTextP(geom_type));
+
     /* check if it is a 2-D point type */
     if (strcmp(geometry_name, "ST_Point") != 0)
         ereport(ERROR,
             (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
             errmsg("%s is not supported", geometry_name)));
     else
     {
-        if ((geom->gflags & FLAG_Z) || (geom->gflags & FLAG_M) || (geom->gflags & (FLAG_Z | FLAG_M)) )
-            ereport(ERROR,
-                (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-                errmsg("Unsupported flags")));
-        else
+        /* For character string, set default SRID to be 4326 for geography datatype */
+        if (!isBinary)
         {
+            InitFunctionCallInfoData(*fcinfo_local, NULL, 2, PG_GET_COLLATION(), NULL, NULL);
             fcinfo_local->args[0].value = geom_datum;
-            /* Flipping the coordinates since Geography Datatype stores the point in Reverse Order i.e. (long, lat)  */
-            if (!isBinary)
-                geom_datum = st_flip_coord_p(fcinfo_local);
+            fcinfo_local->args[1].value = Int32GetDatum(4326);
+            geom_datum = gserialized_set_srid_p(fcinfo_local);
+        }
 
+        /* Flipping the coordinates since Geography Datatype stores the point in Reverse Order i.e. (long, lat) */
+        if (!isBinary)
+        {
+            InitFunctionCallInfoData(*fcinfo_local, NULL, 1, PG_GET_COLLATION(), NULL, NULL);
             fcinfo_local->args[0].value = geom_datum;
-            lat = DatumGetFloat8(lwgeom_x_p(fcinfo_local));
+            geom_datum = st_flip_coord_p(fcinfo_local);
+        }
 
-            /* Checking if latitude falls in allowed range -> [-90.0, 90.0] */
-            if(lat <= 90.0 && lat >= -90.0)
-                PG_RETURN_DATUM(geom_datum);
-            else
-                ereport(ERROR,
-                    (errcode(ERRCODE_DATA_EXCEPTION),
-                    errmsg("Latitude values must be between -90 and 90 degrees")));
+        /* Get latitude */
+        InitFunctionCallInfoData(*fcinfo_local, NULL, 1, PG_GET_COLLATION(), NULL, NULL);
+        fcinfo_local->args[0].value = geom_datum;
+        lat = DatumGetFloat8(lwgeom_x_p(fcinfo_local));
+
+        /* Checking if latitude falls in allowed range -> [-90.0, 90.0] */
+        if(lat <= 90.0 && lat >= -90.0)
+        {
+            PG_RETURN_DATUM(geom_datum);
         }
+        else
+            ereport(ERROR,
+                (errcode(ERRCODE_DATA_EXCEPTION),
+                errmsg("Latitude values must be between -90 and 90 degrees")));
     }
 
-    PG_RETURN_DATUM(geom_datum);
+    /* This point should never be reached, but to satisfy the compiler: */
+    PG_RETURN_NULL();
+}
+
+/* This function stores valid SRIDs for Geography types*/
+Datum
+get_valid_srids(PG_FUNCTION_ARGS)
```

Comment:
Is this function now being called only from C code? if yes then this can simply be a normal function. Even better is to just declare `valid_srids` array statically at the top of this file and directly use it to completely get rid of this function.